Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth
# Mind Control Refactoring

## Secret Changelog:
- Refactored the script of Mind Control events. Improvements:
  - Created 3 new effects: `remove_mind_control_modifier_effect`, `increase_mind_control_stage_effect`, `decrease_mind_control_stage_effect`.
  - Created 1 new trigger: `can_be_mind_controlled_by_scourge_trigger`
  - The event that increases your **Mind Control Modifier** (`WCUND.1106`) and the event that decrease your **Mind Control Modifier** (`WCUND.1107`) can't create a few event windows. For example, when you decrease **Mind Control Variable** by using `event WCUND.1112` console command, the event shall fire at 75 **Mind Control Variable**. If you continue decreasing **Mind Control Variable** without closing the event window, the game shall not open another event window.
- Refactored WCUND.1109. Nothing special, just used `else_if` and `else` and replaced `is_ruler` by `is_landed` because ruler may be unlanded.
- Renamed `mind_control_2` because `mind_control_2` and `mind_control_3` had the same name.

## How to test
Start as the Lich King in the 603 bookmark, use `event WCUND.1111` and `event WCUND.1112` to increase/decrease **Mind Control Variable** a lot of times and see what happens. Try to break it somehow.
